.github/workflows/build.yml

```
@@ -3,8 +3,10 @@ name: build
 on:
   push:
     branches:
-      - 'master'
-  pull_request: {}
+      - 'final'
+  pull_request:
+    branches:
+      - 'final'
```

I feel it is better if we keep the same build file across branches. We don't need to deal with merge conflicts etc 

something like:

```suggestion
      - 'master'
      - 'final'
  pull_request:
    branches:
     - 'master'
     - 'final'
```